src/runtime/types/v5.d.ts

```diff
@@ -0,0 +1,42 @@
+import type { MetaResponsePaginationByOffset, MetaResponsePaginationByPage, PaginationByOffset, PaginationByPage, StrapiLocale } from '.'
+
+export interface Strapi5Error {
+  error: {
+    status: number
+    name: string
+    message: string
+    details: Record<string, unknown>
+  }
+}
+
+export interface Strapi5RequestParams {
+  fields?: Array<string>
+  populate?: string | Array<string> | object
+  sort?: string | Array<string>
+  pagination?: PaginationByOffset | PaginationByPage
+  filters?: Record<string, unknown>
+  publicationState?: 'live' | 'preview'
+  locale?: StrapiLocale
+}
+
+export interface Strapi5SystemFields {
+  documentId: string
+  locale?: StrapiLocale
+}
+
+export type Strapi5ResponseData<T> = Strapi5SystemFields & T
```

Comment:
Not sure you want to remove the `Strapi5Response` as it might be used by users.

Reply:
Removed it as it was duplicate from Strapi5ResponseSingle